woocommerce/woocommerce-android
# [Speedy IPP] Cache IPP onboarding result

Closes: #8642


### Description

The PR adds in-memory caching on the IPP onboarding check result

* We cache only Onboarding Completed State
* Run onboarding check in the background on every app start. Make sure to run it only for the users who successfully used IPP before in the last 30 days. Also, make sure to run the check-in atomic way, so if a user after the app’s restart quickly starts IPP flow we won’t mess up the cache with the results for 2 checks that run in parallel. Fetch only for the selected site!
* We can use it in the memory cache
* We invalidate cache:
  * If any error comes from the SDK or the API during the payment flow
  * Plugin Selection clicked
  * Site change
  * Logout


@samiuelson @AnirudhBhat this is important change so I would like to ask both of you to review this

### Testing instructions

* Open the app and make a payment. Notice that the next payment will go faster because we use the cached value
* If payment (any IPP or TTP) payment was done within the last 30 days, notice that there is no loading on the Payment Hub screen
* Change data on the phone on 31 days and notice that when Hub is open then we do an onboarding check 
* Try different combinations of the cases when the cache is not valid anymore. E.g. open the app with onboarding completed, then go to the web and make changes to make onboarding not completed. Try to make payment. Notice that cache is invalidated in this case and next payment we will fetch the onboarding status
* Do the same, but with logout
* Do the same, but with a site change
* Do the same, but with "plugin selection" click on the hub screen
